### PR TITLE
CTP-3414 Implement re-assessment marks transfer

### PR DESCRIPTION
**Pull request by aydevworks**

### CODE REVIEW

File: classes/external/map_assessment.php [hunk 1]
```diff
@@ -43,6 +43,7 @@ public static function execute_parameters() {
             'sourcetype' => new external_value(PARAM_TEXT, 'Source Type', VALUE_REQUIRED),
             'sourceid' => new external_value(PARAM_INT, 'Source ID', VALUE_REQUIRED),
             'mabid' => new external_value(PARAM_INT, 'Assessment Component ID', VALUE_REQUIRED),
+            'reassess' => new external_value(PARAM_INT, 'Reassessment Flag', VALUE_DEFAULT, 0),
```
Comment (aspark21) [inline classes/external/map_assessment.php:46]: should we make it required?

Would have to be a conscious decision to be fetching one or the other & create an awareness that the reassess option exists

File: tests/behat/behat_sitsgradepush.php [hunk 2]
```diff
@@ -280,6 +285,22 @@ public function the_course_is_regraded(string $shortname): void {
         global $DB;
         $course = $DB->get_record('course', ['shortname' => $shortname]);
         grade_regrade_final_grades($course->id);
+
+        // Wait for the grade item to be updated. Check every 10 seconds, timeout in 300 seconds.
```
Comment (aspark21) [inline tests/behat/behat_sitsgradepush.php:289]: Not sure if I looped you in on that, https://tracker.moodle.org/browse/MDL-70854 has now been integrated, 
https://tracker.moodle.org/browse/MDL-81714 (which we have an early draft of in our build) will make use of that to pull through the recalculation state and improve the wording where the formulae need fixing.

Though both of these will be 4.5+ and we will stay on 4.4 and then upgrade to 5.0(4.6).

We probably will want to revisit that once they have been integrated into core.

Comment (aspark21) [reply tests/behat/behat_sitsgradepush.php:289]: see comment on https://github.com/ucl-isd/moodle-core/commit/e50522508ef26a3a09fb4bcefa85ee09da77d8a9#r144757264

File: dashboard.php [hunk 3]
```diff
@@ -43,24 +46,29 @@
...
 $PAGE->set_url($url);
 $PAGE->set_title(get_string('dashboard:header', 'local_sitsgradepush'));
-$PAGE->set_secondary_navigation(false);
-
-// Set the breadcrumbs.
-$PAGE->navbar->add(get_string('courses'), new moodle_url('/course/index.php'));
-$PAGE->navbar->add($course->fullname, new moodle_url('/course/view.php', ['id' => $courseid]));
-$PAGE->navbar->add(get_string('pluginname', 'local_sitsgradepush'),
-    new moodle_url('/local/sitsgradepush/dashboard.php', $param));
+$PAGE->set_heading(format_string($course->fullname, true, ['context' => $context]));
```
Comment (aspark21) [inline dashboard.php:71]: nice, much simpler. I like it :)

File: apiclients/easikit/classes/requests/pushgrade.php [hunk 4]
```diff
@@ -71,7 +71,7 @@ public function __construct(\stdClass $data) {
      */
     public static function transform_data(\stdClass &$data) {
         // Add required data.
-        $rseq = empty($data->reassessment) ? '0' : $data->srarseq;
+        $rseq = $data->reassessment == 1 ? str_pad($data->srarseq, 3, '0', STR_PAD_LEFT) : '0';
```
Comment (aspark21) [inline apiclients/easikit/classes/requests/pushgrade.php:74]: what is going on here?

Comment (aspark21) [reply apiclients/easikit/classes/requests/pushgrade.php:74]: might be worth adding an inline comment explaining

**Comment by aspark21** [comment]
Have done first pass of code review. Looking good.

File: apiclients/easikit/classes/requests/pushgrade.php (hunk 4 above)
Comment (aspark21) [reply apiclients/easikit/classes/requests/pushgrade.php:77]: 👍

**Comment by aydevworks** [comment]
Changes deployed on 44-clc, it seems working...

**Comment by aydevworks** [comment]
Included in https://github.com/ucl-isd/moodle-local_sitsgradepush/pull/58

**Review by aspark21 — APPROVED** [review]

**Pull request merged**